### PR TITLE
Captions selectors

### PR DESCRIPTION
**Pull request by carocao-msft**
# What
selectors for captions realted components

### CODE REVIEW

File: packages/calling-component-bindings/review/beta/calling-component-bindings.api.md [hunk 1]
```diff
@@ -107,6 +109,23 @@ export type CameraButtonSelector = (state: CallClientState, props: CallingBaseSe
...
+    captions: CaptionInfo[];
+};
+
+// @public
+export type ChangeCaptionLanguageButtonSelector = (state: CallClientState, props: CallingBaseSelectorProps) => {
+    supportedCaptionLanguages: string[];
+    currentCaptionLanguage: string;
+};
+
+// @public
```
Comment (PorterNan) [inline packages/calling-component-bindings/review/beta/calling-component-bindings.api.md:123]: These are all beta, not public

File: packages/calling-component-bindings/review/beta/calling-component-bindings.api.md [hunk 2]
```diff
@@ -251,6 +270,13 @@ export type ScreenShareButtonSelector = (state: CallClientState, props: CallingB
 // @public
 export const screenShareButtonSelector: ScreenShareButtonSelector;
 
+// @public
```
Comment (PorterNan) [inline packages/calling-component-bindings/review/beta/calling-component-bindings.api.md:273]: beta

File: packages/calling-component-bindings/review/beta/calling-component-bindings.api.md [hunk 3]
```diff
@@ -251,6 +270,13 @@ export type ScreenShareButtonSelector = (state: CallClientState, props: CallingB
...
+export type StartCaptionsButtonSelector = (state: CallClientState, props: CallingBaseSelectorProps) => {
+    isCaptionsFeatureActive: boolean;
+    currentCaptionLanguage: string;
```
Comment (PorterNan) [inline packages/calling-component-bindings/review/beta/calling-component-bindings.api.md:276]: So the pop up window will be part of the button implementation?

File: packages/calling-component-bindings/src/baseSelectors.ts [hunk 4]
```diff
@@ -139,3 +140,51 @@ export const getParticipantCount = (state: CallClientState, props: CallingBaseSe
...
+/** @private */
+export const getCaptions = (state: CallClientState, props: CallingBaseSelectorProps): CaptionsInfo[] | undefined => {
+  return state.calls[props.callId]?.captionsFeature.captions;
+  return undefined;
+};
+
+/** @private */
+export const getCaptionsStatus = (state: CallClientState, props: CallingBaseSelectorProps): boolean | undefined => {
+  return state.calls[props.callId]?.captionsFeature.isCaptionsFeatureActive;
+  return undefined;
```
Comment (PorterNan) [inline packages/calling-component-bindings/src/baseSelectors.ts:153]: This undefined is not useful when there is no conditional compilation

File: packages/calling-component-bindings/src/captionsSelector.ts [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+  getSupportedCaptionLanguages,
+  getSupportedSpokenLanguages
+} from './baseSelectors';
+import * as reselect from 'reselect';
+import { toFlatCommunicationIdentifier } from '@internal/acs-ui-common';
+import { CaptionInfo } from '@internal/react-components';
+
+/**
+ * Selector type for the {@link StartCaptionsButton} component.
+ * @public
```
Comment (PorterNan) [inline packages/calling-component-bindings/src/captionsSelector.ts:20]: beta

File: packages/calling-component-bindings/src/captionsSelector.ts [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+  state: CallClientState,
+  props: CallingBaseSelectorProps
+) => {
+  captions: CaptionInfo[];
+};
+
+/**
+ * Selector for {@link CaptionsBanner} component.
+ *
+ * @private
```
Comment (PorterNan) [inline packages/calling-component-bindings/src/captionsSelector.ts:115]: Should be all beta, since we are going to usePropsFor these selectors I guess

**Review by PorterNan — APPROVED** [review]
LGTM, just don't forget mark most of them as @beta

File: packages/calling-component-bindings/review/beta/calling-component-bindings.api.md (hunk 3 above)
Comment (carocao-msft) [reply packages/calling-component-bindings/review/beta/calling-component-bindings.api.md:276]: Will need to include current languages when starting captions

**Pull request merged**